src/main/java/seedu/address/logic/commands/FindCommand.java

```diff
@@ -117,59 +162,24 @@ public CommandResult executeIndexSearch(Model model) {
      * @param model The model which contains the list to be searched on.
      * @throws CommandException When the index inputted is not within range.
      */
-    public void checkIndex(Model model) throws CommandException {
+    private void checkIndex(Model model) throws CommandException {
         int personListSize = model.getFilteredPersonList().size();
         if (index > personListSize - 1) { // -1 so that index starts from 0
             throw new CommandException(Messages.MESSAGE_INVALID_PERSON_DISPLAYED_INDEX);
         }
     }
 
-    /**
-     * Returns the index of the FindCommand object.
-     *
-     * @return index
-     */
-    public int getIndex() {
-        return this.index;
-    }
-
-    /**
-     * Returns the namePredicate of the FindCommand object.
-     *
-     * @return namePredicate
-     */
-    public NameContainsKeywordsPredicate getNamePredicate() {
-        return this.namePredicate;
-    }
 
     @Override
     public boolean equals(Object other) {
+        if (other == null) {
+            return false;
+        }
         return other == this // short circuit if same object
                 || (other instanceof FindCommand // instanceof handles nulls
-                && ((findByNameIsEquals((FindCommand) other)) || findByIndexIsEquals((FindCommand) other)));
-    }
-
-    /**
-     * Checks if another FindCommand object which searches by name is equal to the current FindCommand object.
-     *
-     * @param otherFind The other FindCommand object to be checked
-     * @return Whether the otherFind is equal to this
-     */
-    public boolean findByNameIsEquals(FindCommand otherFind) {
-        return (otherFind.getIndex() == INVALID_INDEX && this.index == INVALID_INDEX)
-                && (otherFind.getNamePredicate() != null && this.namePredicate != null)
-                && (this.namePredicate.equals(otherFind.getNamePredicate()));
+                && ((FindCommand) other).namePredicate.equals(this.namePredicate)
+                && ((FindCommand) other).index == index
+                && predicate.equals(((FindCommand) other).predicate));
```

Comment:
Maybe can consider `FindCommand otherCommand = (FindCommand) other;` first?

Reply:
I do not think it is necessary for 3 field accesses.